### ISSUE TITLE
FilteringGeneratorDelegate does not override writeStartObject(Object forValue)

### ISSUE THREAD

**Issue by arnaudroger** [FasterXML/jackson-core#359]
that leads to 

```
Caused by: java.lang.NullPointerException
	at com.fasterxml.jackson.core.filter.FilteringGeneratorDelegate.writeFieldName(FilteringGeneratorDelegate.java:278)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:670)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:690)
	... 43 more
```

when using the BeanSerializer

writeStartObject(Object forValue) should do the same as writeStartObject().

will submit a PR

**Comment by arnaudroger** [comment]
see https://github.com/FasterXML/jackson-core/pull/360

**Comment by cowtowncoder** [comment]
Merged for 2.8.8 / 2.9.0.pr2
